xscen/aggregate.py

```diff
@@ -294,6 +294,7 @@ def spatial_mean(
     ds: xr.Dataset,
     method: str,
     *,
+    spatial_subset: bool = False,
     call_clisops: bool = False,
     region: dict = None,
```

Comment:
```suggestion
    spatial_subset: bool = None,
    call_clisops: bool = False,
    region: dict = None,
```
Suggestion to do the following:

If `spatial_subset` is None, `region` is not None and `method` is in `['cos-lat', 'mean'], then `spatial_subset=True`.

For methods "xESMF" and "interp_centroid", the information of `region` is used for the spatial mean : subsetting is implicit. But with the two other methods there is an ambiguity.  I just had this problem where I assumed that `spatial_mean` was taking the average over my given region, but wasn't passing `call_clisops=True` so it was taking the average over all the domain. It doesn't make sense to me to accept a `region` arg and just ignore it silently like this.

Reply:
I fear that dynamically changing the behaviour of `None` might be confusing? An alternative that would have worked in your example would be to raise an error if those 3 conditions are met.

Reply:
On the contrary, I feel that a default of None is much more like a "unset" which means the default behaviour depends on something else. 
In our case here, as discussed earlier, we could have chosen to always subset, but that _might_ introduce some bugs in some edge cases (subsetting without buffer before averaging with xesmf, for example).

Reply:
When you have time, could you check the new commit to make sure that my implementation is what you had in mind?